store/tikv/client.go

```diff
@@ -86,13 +89,114 @@ type connArray struct {
 	v     []*grpc.ClientConn
 	// Bind with a background goroutine to process coprocessor streaming timeout.
 	streamTimeout chan *tikvrpc.Lease
+
+	// For batch commands.
+	batchCommandsCh        chan *batchCommandsEntry
+	batchCommandsClients   []*batchCommandsClient
+	tikvTransportLayerLoad uint64
+}
+
+type batchCommandsClient struct {
+	conn                   *grpc.ClientConn
+	client                 tikvpb.Tikv_BatchCommandsClient
+	batched                sync.Map
+	idAlloc                uint64
+	tikvTransportLayerLoad *uint64
+
+	// Indicates the batch client is closed explicitly or not.
+	closed int32
+	// Protect client when re-create the streaming.
+	clientLock sync.Mutex
+}
+
+func (c *batchCommandsClient) isStopped() bool {
+	return atomic.LoadInt32(&c.closed) != 0
+}
+
+func (c *batchCommandsClient) failPendingRequests(err error) {
+	c.batched.Range(func(key, value interface{}) bool {
+		id, _ := key.(uint64)
+		entry, _ := value.(*batchCommandsEntry)
+		entry.err = err
+		close(entry.res)
+		c.batched.Delete(id)
+		return true
+	})
+}
+
+func (c *batchCommandsClient) batchRecvLoop(cfg config.TiKVClient) {
+	defer func() {
+		if r := recover(); r != nil {
```

Comment:
@hicqu how about batchSendLoop/batchRecvLoop goroutine exit but the client still send the request to TiDB, maybe we should keep loop continue running